app/src/main/java/org/mozilla/fenix/StrictModeManager.kt

```diff
@@ -2,25 +2,58 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+// This class implements the alternative ways to suppress StrictMode with performance
+// monitoring by wrapping the raw methods. This lint check tells us not to use the raw
+// methods so we suppress the check.
+@file:Suppress("MozillaStrictModeSuppression")
+
 package org.mozilla.fenix
 
 import android.os.Build
+import android.os.Looper
 import android.os.StrictMode
+import androidx.annotation.VisibleForTesting
+import androidx.annotation.VisibleForTesting.PRIVATE
 import androidx.fragment.app.Fragment
 import androidx.fragment.app.FragmentManager
 import mozilla.components.support.ktx.android.os.resetAfter
+import org.mozilla.fenix.components.Components
+import org.mozilla.fenix.perf.Performance
 
 private const val MANUFACTURE_HUAWEI: String = "HUAWEI"
 private const val MANUFACTURE_ONE_PLUS: String = "OnePlus"
 
 /**
  * Manages strict mode settings for the application.
  */
-class StrictModeManager(config: Config) {
+class StrictModeManager(
+    config: Config,
+
+    // Ideally, we'd pass in a more specific value but there is a circular dependency: StrictMode
+    // is passed into Core but we'd need to pass in Core here. Instead, we take components and later
+    // fetch the value we need from it.
+    //
+    // Public to be accessible by inline functions.
+    val components: Components
```

Comment:
I was looking at `Components.kt` and what the other components seem to do is just pass `context`. Should we do that here and access `Profiler` from it?  I can see how that would be redundant though since `context.components....` would be similar to just passing `this` from components.

Reply:
I think that's an anti-pattern. With dependency injection style patterns, you should pass in the most specific object you can. The problem with passing in objects with larger scope, is that:
1) you give the class access to a large number of classes. By having access to a wider range of classes, your class can (in theory) change much more about the program. By limiting the objects that are passed in, it's easier for devs to make analyses when looking at the code.
2) your test code is more fragile because it relies on accessing objects several levels down and any of the upper layers can be re-factored
3) ^ for similar reasons, your code is less explicit
4) passing in context more closely ties your code to the Android framework and the Android framework is difficult to test